tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -2109,13 +2177,18 @@ static unsigned resolveFromLineCol(unsigned Line, unsigned Col,
 
 static llvm::StringMap<llvm::MemoryBuffer*> Buffers;
 
-static llvm::MemoryBuffer *getBufferForFilename(StringRef Filename) {
+static llvm::MemoryBuffer *
+getBufferForFilename(StringRef Filename,
+                     const llvm::StringMap<std::string> &VFSFiles) {
+  auto VFSFileIt = VFSFiles.find(Filename);
+  if (VFSFileIt != VFSFiles.end())
+    Filename = VFSFileIt->second;
```

Comment:
Not sure how idiomatic it is to overwrite a function parameter. This was certainly surprising to me, but I don't have much experience with C++ yet.

Reply:
I think that usually you don't. But maybe it's idiomatic to do it in cases like this where you're just "setting up" parameters in the beginning. Not sure. I'll change it to not mutate to be less surprising.